src/Controllers/LinkFieldController.php

```
@@ -136,15 +138,29 @@ public function linkDelete(): HTTPResponse
         if (!SecurityToken::inst()->checkRequest($this->getRequest())) {
             $this->jsonError(400);
         }
-        // delete() will also delete any published version immediately
-        $link->delete();
+        if ($link->hasExtension(Versioned::class)) {
+            $link->doArchive();
+        } else {
+            $link->delete();
+        }
         // Update owner object if this Link is on a has_one relation on the owner
         $owner = $this->getOwnerFromRequest();
         $ownerRelation = $this->getOwnerRelationFromRequest();
         $hasOne = Injector::inst()->get($owner->ClassName)->hasOne();
         if (array_key_exists($ownerRelation, $hasOne) && $owner->canEdit()) {
+            // Update draft owner table
             $owner->$ownerRelation = null;
             $owner->write();
+            // Update live owner table if owner is versioned
+            if ($owner->hasExtension(Versioned::class)) {
+                $draftTable = $owner->getSchema()->tableForField($owner::class, $ownerRelation . 'ID');
+                $liveTable = $draftTable . '_Live';
+                SQLUpdate::create()
+                    ->setTable($liveTable)
+                    ->assign('"' . $ownerRelation . 'ID"', 0)
+                    ->addWhere(['ID' => $owner->ID])
+                    ->execute();
```

This feels a bit hacky.
It's also not consistent with what happens when you archive related records in other situations. I don't think we should be doing this part.

The ORM already knows how to handle the scenario where a related record doesn't exist anymore. We don't need to try to handle that here.

If the old ID is causing errors in the UI, we need to instead swallow those errors (an outdated ID is fine, we just treat is as though the ID wasn't there in literally every other scenario in the CMS)

---

Should have realised this last time, sorry about that.